### DISCUSSION TITLE
Run scripts from Stream Deck with Script Kit

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#101] [Share]
https://user-images.githubusercontent.com/36073/113206178-61aee400-922c-11eb-8c62-573f8beca69d.mp4

You can find a "Prepare Script for Stream Deck" command in the main menu:

![CleanShot 2021-09-14 at 07 45 45](https://user-images.githubusercontent.com/36073/133269177-18b1ddec-21c1-4b2e-9d12-903ea77bc072.png)

(the code examples below are a bit out-dated)

**Comment by lannonbr** [comment]
I wonder if it would be useful to make a "export to streamdeck" script that you pick an existing script and wrap it in a shell script and then copy the output path to the clipboard to paste onto a button in the streamdeck software.

**Reply by lannonbr** [reply]
And done:

```js
// Menu: Export script to StreamDeck

let script = await arg(`What script would you like to export`,
    async () => (await cli("scripts")).scripts,
)

let mode = await arg('Would you like to open the Script Kit Prompt or run the script directly on launch?', [
    { name: 'Directly' , description: 'run with ~/.kit/script', value: 'script' },
    { name: 'Prompt', description: 'run with ~/.kit/kar', value: 'kar' }
])

let shellScriptPath = path.join(kenvPath("deck"), `${script.split(".js")[0]}.sh`)
writeFile(shellScriptPath, `~/.kit/${mode} ~/.kenv/scripts/${script}\n`)

chmod(755, shellScriptPath)

copy(shellScriptPath)
```

**Reply by johnlindquist** [reply]
Yah, main menu will eventually have an "export" option. Webhooks, compiled binary, and others are all reasonable.

Here's a script that will generate an .sh for you in the meantime:

> Note: Running this from the main menu has a minor menu-caching bug I'll fix tonight. So run from a shortcut until tomorrow :)

[Click to install "Wrap Script for Stream Deck"](https://scriptkit.app/api/new?name=wrap-script-for-stream-deck&url=https://gist.githubusercontent.com/johnlindquist/341cf67898b008b123e74ebf6d746f2a/raw/c24fcefa5b6dfae2a8edbaa7f1f4e7272a939a9e/wrap-script-for-stream-deck)
```js
//Menu: Wrap Script for Stream Deck
//Description: Creates a .sh file around a script
//Author: John Lindquist
//Twitter: @johnlindquist
//Shortcut: opt h

let { menu, findScript } = await cli("fns")

let createCommand = (launchApp, script) =>
  launchApp
    ? `~/.kit/kar ${script}`
    : `~/.kit/script ~/.kenv/scripts/${script}.js`

let script = await arg(
  "Which script do you want to wrap?",
  menu
)

let launchApp = await arg("Launch app?", [
  {
    name: "Yes",
    value: true,
    description: ".sh that opens the prompt",
  },
  {
    name: "No",
    value: false,
    description: ".sh that runs in the background",
  },
])

let name = await arg({
  message: "Name your .sh",
  hint: '"hello-world" would create "hello-world.sh"',
})

let binPath = kenvPath("deck", name + ".sh")
mkdir("-p", path.dirname(binPath))
await writeFile(binPath, createCommand(launchApp, script))
chmod(755, binPath)
edit(binPath)
let resolvedPath = path.resolve(binPath)
copy(resolvedPath)

await arg(
  `Paste into Stream Deck`,
  md(`
${resolvedPath}

copied to your clipboard!

Hit "Enter" to launch Stream Deck

`)
)

exec(`open -a "Stream Deck.app"`)
```

**Reply by johnlindquist** [reply]
@lannonbr lol, looks like you beat me to it 😂

**Reply by JamesHenry** [reply]
I hadn't tried running a script from stream deck for quite a while and for me the above no longer works but it's hard to try and piece together what breaking changes might have occurred between recent betas.

Hoping this might be a quick one for you @johnlindquist?

I simply installed your script unmodified and get:

![image](https://user-images.githubusercontent.com/900523/133238952-a18cec2b-b06b-4041-b1e5-116a41c860df.png)

**Reply by johnlindquist** [reply]
@JamesHenry 

Try the command from the main menu (I updated the main post to reflect it). 

![CleanShot 2021-09-14 at 07 45 45](https://user-images.githubusercontent.com/36073/133269177-18b1ddec-21c1-4b2e-9d12-903ea77bc072.png)

**Reply by JamesHenry** [reply]
Thank @johnlindquist but I think there's an issue with it - it doesn't seem to support custom kenvs currently, in terms of the contents of the `.sh` file it just assumes that the script lives in `~/.kenv/scripts/` which in my case it does not, it is coming from my kenvs directory

**Reply by johnlindquist** [reply]
Ah, good catch. I'll get that fixed!
